src/measurement/position_measurement.js

```diff
@@ -427,7 +427,7 @@ export function coordsChar(cm, x, y) {
 function wrappedLineExtent(cm, lineObj, preparedMeasure, y) {
   let measure = ch => intoCoordSystem(cm, lineObj, measureCharPrepared(cm, preparedMeasure, ch), "line")
   let end = lineObj.text.length
-  let begin = findFirst(ch => measure(ch).bottom < y, end - 1, 0) + 1
+  let begin = findFirst(ch => measure(ch).bottom <= y, end - 1, -1) + 1
```

Comment:
Ah, makes sense. So this is just a somewhat kludgy way to make sure `begin` can end up being 0? Any chance we can express the search predicate in such a way that the two `-1`'s and the final `+1` disappear?

(I'm deeply suspicious of code that's littered with +/- 1's, since it breeds off-by-one errors -- which I think you mentioned you had a lot of.)